src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java

```diff
@@ -523,7 +528,13 @@ public static boolean uploadRepo(String repoPath, String repoName, String[] file
         payload.put("platform", CommonUtils.getOS());
 
         codeSyncProgressIndicator.setMileStone(InitRepoMilestones.PROCESS_RESPONSE);
-        JSONObject response = codeSyncClient.uploadRepo(accessToken, payload);
+        JSONObject response = null;
+        try{
+            response = codeSyncClient.uploadRepo(accessToken, payload);
```

Comment:
There are try-catch inside uploadRepo, what excpetion are you catching here @syedgulahmed?

Reply:
Yeah, there are try-catch block in codeSyncClient.uploadRepo but I can't change state in that method.

Reply:
Alright, what excpetion are you catching here then @syedgulahmed?
I think we need to handle each catch block inside uploadRepo explictly. Or maybe return True/False from uploadRepo and check its output to set `syncInProcess = false/true`, thoughts?

Reply:
As mentioned above, I do not think these are needed.